Warning when compiling benchmarks using MSVC

**Describe the bug**
When I compile a benchmark I get a C4880 warning from MSVC:
> warning C4880: Implicit conversion from 'T *const ' to bool. Possible information loss

**Expected behavior**
No warnings.

**Reproduction steps**
The following minimal example demonstrates the problem:
```cpp
TEST_CASE("bench", "[bench]")
{
    BENCHMARK("nothing")
    {
        return 0;
    };
}
```


**Platform information:**
 - OS: **Windows 11**
 - Compiler+version: **MSVC v14.44 (Visual Studio 17.14)**
 - Catch version: **any**


**Additional context**
N/A

--- a/src/catch2/internal/catch_unique_ptr.hpp
+++ b/src/catch2/internal/catch_unique_ptr.hpp
@@ -92,7 +92,7 @@ namespace Detail {
         }
 
         explicit operator bool() const {
-            return m_ptr;
+            return m_ptr != nullptr;
         }
 
         friend void swap(unique_ptr& lhs, unique_ptr& rhs) {

